### PR TITLE
[ticket/16204] Remove hooks system

### PR DESCRIPTION
**Pull request by marc1706**
PHPBB3-16204

Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16204

### CODE REVIEW

File: phpBB/common.php [hunk 1]
```diff
@@ -143,10 +143,6 @@
 	require_once($phpbb_root_path . $config['exts_composer_vendor_dir'] . '/autoload.php');
 }
 
-// Add own hook handler
-require($phpbb_root_path . 'includes/hooks/index.' . $phpEx);
-$phpbb_hook = new phpbb_hook(array('exit_handler', 'phpbb_user_session_handler', 'append_sid', array('template', 'display')));
-
 /* @var $phpbb_hook_finder \phpbb\hook\finder */
 $phpbb_hook_finder = $phpbb_container->get('hook_finder');
```
Comment (rubencm) [inline phpBB/common.php]: This can't be removed??

Comment (marc1706) [reply phpBB/common.php]: It can, missed that apparently.

File: phpBB/includes/functions.php [hunk 2]
```diff
@@ -4543,17 +4531,31 @@ function garbage_collection()
 * This function supports hooks.
```
Comment (rubencm) [inline phpBB/includes/functions.php]: `* This function supports hooks.`

File: phpBB/includes/functions.php [hunk 3]
```diff
@@ -1514,7 +1514,7 @@ function tracking_unserialize($string, $max_depth = 3)
 */
```
Comment (rubencm) [inline phpBB/includes/functions.php]: `* This function supports hooks.`

In the function documentation

**Pull request merged**